src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubectlBuildWrapper.java

```diff
@@ -87,132 +67,31 @@ public String getCaCertificate() {
 
         context.env("KUBECONFIG", configFile.getRemote());
         context.setDisposer(new CleanupDisposer(tempFiles));
 
-        String tlsConfig;
-        if (caCertificate != null && !caCertificate.isEmpty()) {
-            FilePath caCrtFile = workspace.createTempFile("cert-auth", "crt");
-            String ca = caCertificate;
-            if (!ca.startsWith(BEGIN_CERTIFICATE)) {
-                ca = wrapWithMarker(BEGIN_CERTIFICATE, END_CERTIFICATE, ca);
-            }
-            caCrtFile.write(ca, null);
-            tempFiles.add(caCrtFile.getRemote());
-
-            tlsConfig = " --certificate-authority=" + caCrtFile.getRemote();
-        } else {
-            tlsConfig = " --insecure-skip-tls-verify=true";
-        }
-
-        int status = launcher.launch()
-                .cmdAsSingleString("kubectl config --kubeconfig=\"" + configFile.getRemote()
-                        + "\" set-cluster k8s --server=" + serverUrl + tlsConfig)
-                .join();
-        if (status != 0) throw new IOException("Failed to run kubectl config "+status);
-
-        final StandardCredentials c = getCredentials();
-
-        String login;
-        if (c == null) {
-            throw new AbortException("No credentials defined to setup Kubernetes CLI");
+        StandardCredentials credentials = CredentialsProvider.findCredentialById(credentialsId, StandardCredentials.class, build, Collections.emptyList());
+        if (credentials == null) {
+            throw new AbortException("No credentials found for id \"" + credentialsId + "\"");
         }
-
-        if (c instanceof FileCredentials) {
-            try (InputStream in = ((FileCredentials) c).getContent(); OutputStream out = configFile.write()) {
-                IOUtils.copy(in, out);
-            }
-            return;
+        KubernetesAuth auth = AuthenticationTokens.convert(KubernetesAuth.class, credentials);
+        if (auth == null) {
+            throw new AbortException("Unsupported Credentials type " + credentials.getClass().getName());
         }
-
-        if (c instanceof StringCredentials) {
-            login = "--token=" + ((StringCredentials) c).getSecret().getPlainText();
-        } else if (c instanceof TokenProducer) {
-            login = "--token=" + ((TokenProducer) c).getToken(serverUrl, null, true);
-        } else if (c instanceof UsernamePasswordCredentials) {
-            UsernamePasswordCredentials upc = (UsernamePasswordCredentials) c;
-            login = "--username=" + upc.getUsername() + " --password=" + Secret.toString(upc.getPassword());
-        } else if (c instanceof StandardCertificateCredentials) {
-            StandardCertificateCredentials scc = (StandardCertificateCredentials) c;
-            KeyStore keyStore = scc.getKeyStore();
-            String alias;
+        // create Kubeconfig
+        try (Writer w = new OutputStreamWriter(new FileOutputStream(configFile.getRemote()), "UTF-8")) {
```

Comment:
JENKINS-61239: this is wrong, cannot work with agents. Use `configFile.write()`.